src/common/config_parser.cpp

```diff
@@ -159,9 +164,11 @@ void ConfigParser::addOptionsGeneral(cli::CLIWrapper& cli) {
 void ConfigParser::addOptionsServer(cli::CLIWrapper& cli) {
   // clang-format off
   auto previous_group = cli.switchGroup("Server options");
+  // TODO why is this needed?
+  size_t defaultPort = mode_ == cli::mode::selfadaptive ? 0 : 8080;
```

Comment:
It seems that you want to run marian-adaptive in the CLI mode by default. This happens if the port number is set to `0`. Can you remove TODO and put a short comment mentioning this instead?

Reply:
These lines are gone now that adaptive server is a separate executable